arcaflow_plugin_coremark_pro/coremark_pro_plugin.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+import subprocess
+import sys
+import typing
+from arcaflow_plugin_sdk import plugin
+from coremark_pro_schema import (
+    CertifyAllParams,
+    certifyAllResultSchema,
+    SuccessOutput,
+    ErrorOutput,
+)
+
+
+def run_oneshot_cmd(command_list, workdir) -> str:
+    try:
+        cmd_out = subprocess.check_output(
+            command_list, stderr=subprocess.PIPE, text=True, cwd=workdir
+        )
+    except subprocess.CalledProcessError as error:
+        return "error", ErrorOutput(
+            f"{error.cmd[0]} failed with return code "
+            f"{error.returncode}:\n{error.output}"
+        )
+    return "completed", cmd_out
+
+
+@plugin.step(
+    id="certify-all",
+    name="certify-all",
+    description=(
+        "Runs all of the nine tests, collects their output scores, and processes them "
+        "to generate the final CoreMark-PRO score"
+    ),
+    outputs={"success": SuccessOutput, "error": ErrorOutput},
+)
+def certify_all(
+    params: CertifyAllParams,
+) -> typing.Tuple[str, typing.Union[SuccessOutput, ErrorOutput]]:
+
+    ca_cmd = [
+        "make",
+        "-s",
+        "certify-all",
+        f"XCMD='-c{params.contexts} -w{params.workers}'",
+    ]
+
+    ca_return = run_oneshot_cmd(ca_cmd, "/root/coremark-pro")
+
+    if ca_return[0] == "error":
+        return ca_return
+
+    ca_results = {
+        "cjpeg-rose7-preset": {},
+        "core": {},
+        "linear_alg-mid-100x100-sp": {},
+        "loops-all-mid-10k-sp": {},
+        "nnet_test": {},
+        "parser-125k": {},
+        "radix2-big-64k": {},
+        "sha-test": {},
+        "zip-test": {},
+        "CoreMark-PRO": {},
+    }
+
+    for line in ca_return[1].splitlines():
+        try:
+            line_name = line.split()[0]
```

Comment:
Repeating the `line.split()` seems excessive: a local variable would be good here.